tests/unit/fsattrs_test.c

```diff
@@ -0,0 +1,46 @@
+#include <test.h>
+#include <cmockery.h>
+#include <fsattrs.h>
+#include <stdlib.h>
+
+
+static void test_file_immutable_flag(void)
+{
+    /* Get immutable flag from non existent file */
+    unlink("no_such_file_639123");
+    bool flag;
+    FSAttrsResult res = FileGetImmutableFlag("no_such_file_639123", &flag);
+    assert_int_not_equal(res, FS_ATTRS_SUCCESS);
+    assert_int_not_equal(res, FS_ATTRS_FAILURE);
```

Comment:
These asserts are weird, especially without a comment. Are you trying to check that the result is `FS_ATTRS_NOENTRY` or `FS_ATTRS_NOTSUPP`? I'd go for the more readable:

```suggestion
    FSAttrsResult res = FileGetImmutableFlag("no_such_file_639123", &flag);
    assert(res == FS_ATTRS_NOENTRY || res == FS_ATTRS_NOTSUPP);
```

That also seems more correct to me if we in the future add more possible values to the enum.

Reply:
Then you wont know if it was `FS_ATTRS_SUCCESS` or `FS_ATTRS_FAILURE` that triggered the assert. But either way is fine by me.